nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/AbstractOperationMultiValuedProperty.java

```diff
@@ -45,9 +46,13 @@ protected void checkFieldType(Type type, Object value) throws OperationException
             throw new UnsupportedOperationException("Manage only lists of scalar items");
         }
 
-        if (!itemType.newInstance().getClass().equals(value.getClass())) {
-            throw new UnsupportedOperationException(String.format("Given type \"%s\" value is not a %s type", value,
-                    itemType.getName()));
+        String exceptionReason = String.format("Given type \"%s\" value is not a %s type", value, itemType.getName());
```

Comment:
There's no need to compute the error message in the general case when there's no failure, so please move this (duplicating if needed) a few lines down to where it's actually needed.